pilot/pkg/security/authz/model/model_test.go

```diff
@@ -458,3 +459,49 @@ func TestModel_Generate(t *testing.T) {
 		}
 	}
 }
+
+func TestGetUsersIncludingAliases(t *testing.T) {
+	type inStruct struct {
+		trustDomainAliases []string
+		users              []string
+	}
+	testCases := []struct {
+		name   string
+		in     inStruct
+		expect []string
+	}{
+		{
+			name:   "No trust domain aliases",
+			in:     inStruct{nil, []string{"cluster.local/ns/foo/sa/bar"}},
+			expect: []string{"cluster.local/ns/foo/sa/bar"},
+		},
+		{
+			name:   "One trust domain alias, one principal",
+			in:     inStruct{[]string{"td1"}, []string{"cluster.local/ns/foo/sa/bar"}},
+			expect: []string{"cluster.local/ns/foo/sa/bar", "td1/ns/foo/sa/bar"},
+		},
+		{
+			name:   "One trust domain alias, two principals",
+			in:     inStruct{[]string{"td1"}, []string{"cluster.local/ns/foo/sa/bar", "cluster.local/ns/yyy/sa/zzz"}},
+			expect: []string{"cluster.local/ns/foo/sa/bar", "cluster.local/ns/yyy/sa/zzz", "td1/ns/foo/sa/bar", "td1/ns/yyy/sa/zzz"},
+		},
+		{
+			name:   "Two trust domain aliases, one principal",
+			in:     inStruct{[]string{"td1", "td2"}, []string{"cluster.local/ns/foo/sa/bar"}},
+			expect: []string{"cluster.local/ns/foo/sa/bar", "td1/ns/foo/sa/bar", "td2/ns/foo/sa/bar"},
+		},
+		{
+			name: "Two trust domain aliases, two principals",
+			in:   inStruct{[]string{"td1", "td2"}, []string{"cluster.local/ns/foo/sa/bar", "cluster.local/ns/yyy/sa/zzz"}},
+			expect: []string{"cluster.local/ns/foo/sa/bar", "cluster.local/ns/yyy/sa/zzz",
```

Comment:
Same here. Only expect the strings starting with "td1" and "td2".

Reply:
Same as above.